CoreGraphFile/src/au/gov/asd/tac/constellation/graph/file/GraphObjectUtilities.java

```diff
@@ -61,15 +85,54 @@
      */
     public static GraphDataObject createMemoryDataObject(final String name, final boolean numbered) {
         GraphDataObject gdo = null;
+        final FileObject root = FILE_SYSTEM.getRoot();
         try {
-            final String fnam = numbered ? String.format("%s%d%s", name, ++fileCounter, GraphDataObject.FILE_EXTENSION) : String.format("%s%s", name, GraphDataObject.FILE_EXTENSION);
-            final FileObject root = FILE_SYSTEM.getRoot();
+            String fnam = graphObjectUtilities.getNewFileName(name, numbered, root);
+            while (graphObjectUtilities.isFileNameDuplicateInMemory(fnam, root)) {
+                //If after all of the above the filename already exists in memory, start again.
+                fnam = graphObjectUtilities.getNewFileName(fnam, numbered, root);
+            }
+            while (fnam.length() >= FILENAME_LENGTH_LIMIT) {
+                fnam = (String) JOptionPane.showInputDialog(null,
+                        CHOOSE_FILENAME,
+                        FILENAME_TITLE,
+                        JOptionPane.INFORMATION_MESSAGE,
+                        null,
+                        null,
+                        fnam);
+            }
+            fnam = String.format("%s%s", fnam, GraphDataObject.FILE_EXTENSION);
             final FileObject fo = FileUtil.createData(root, fnam);
             gdo = (GraphDataObject) DataObject.find(fo);
         } catch (IOException ex) {
             Exceptions.printStackTrace(ex);
         }
-
         return gdo;
     }
+
+    private String getNewFileName(final String name, final boolean numbered, final FileObject root) {
+        final List<FileObject> files = Arrays.stream(root.getChildren()).filter(file -> file.getName().equals(name)).collect(Collectors.toList());
+        if (files.size() > 0) {
+            if (name.endsWith(COPY_STRING)) {
+                return String.format("%s (%d)", name, 1);
+            }
+
+            final Matcher matcher = Pattern.compile(COPY_STRING_PATTERN).matcher(name);
+            //final Matcher matcher =  COPY_NAME_MATCHER.matcher(name);
```

Comment:
if you move the instantiation of `COPY_NAME_MATCHER` out of the constructor, you should be able to uncomment this line and remove the line above it

Reply:
My bad, I already had it compiled. Bad debug left in the code. Removed :)